src/client/Microsoft.Identity.Client/ManagedIdentity/ManagedIdentityClient.cs

```diff
@@ -22,6 +23,10 @@ internal class ManagedIdentityClient
         private const string LinuxHimdsFilePath = "/opt/azcmagent/bin/himds";
         internal static ManagedIdentitySource s_sourceName = ManagedIdentitySource.None;
 
+        // Holds the most recently minted mTLS binding certificate for this application instance.
+        private X509Certificate2 _runtimeMtlsBindingCertificate;
+        internal X509Certificate2 RuntimeMtlsBindingCertificate => Volatile.Read(ref _runtimeMtlsBindingCertificate);
```

Comment:
I understand it's a non-goal to rotate certs etc. But this will return expired certificates.

Reply:
Yes, it can. The cert rotation logic PR will handle this